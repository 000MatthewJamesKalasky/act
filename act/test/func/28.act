function f (int<4> x) : int<5>
{
  int<3> w;
  chan!(int) O;

  chp {
     w := x + x;
     O!w;
     self := w
  }
}


defproc test()
{
   int x;

   chp {
     x := f(x)
   }
}

test t;
