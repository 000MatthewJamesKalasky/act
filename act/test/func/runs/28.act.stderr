ERROR: File `28.act', line 4, col 7
	Declaration for `O': function body can only declare data types!
